### PR TITLE
Fix documentation issues

### LINKED ISSUE #34: Document how to do release for software team
The release can be done through PR as well--but some steps need to be completed on the releaser's local copy and pushed. 

1. locally use git flow start release X.X.X to open a release branch. 
2. update the RELEASENOTES.md document and push branch to GitHub
3. create a PR that has a base branch of master of the branch and document all the changes in the PR comment. 
4. if the checks pass and we're happy with the release, merge the PR
5. On the local copy do git flow finish release X.X.X 
6. push the local copy of develop
7. push the version numbers
8. download a tar.gz and update the [Zenodo entry](https://zenodo.org/record/5579858#.YXQsrHlMHOQ) with the new version.

### LINKED ISSUE #53: Inconsistency with documentation on lifetime units in levelfiles
Found an inconsistency in [the documentation on levelfiles](https://nrcascadesim.readthedocs.io/en/latest/02_Executables.html?highlight=levelfile#levelfile-input-format):

`100000000000000.0 (1e+14 as, or 1 ms)`

I am not sure which is true - if the units of the lifetimes are attoseconds, or if the 1e+14 entry is equivalent to 1 ms. They cannot both be true, as 1e+14 as is only equivalent to 0.1 ms. It seems more likely that the attosecond unit is correct, but we should confirm this before fixing the docs.

### PR DESCRIPTION
**Pull request by gerudo7**
Not meant to resolve all github issues with documentation - the primary purpose was to resolve #53 and the documentation issue discussed in #48, but I included some other improvements as well.
* Resolve #53 by changing lifetime references of units to those used in the code and levelfiles (femtoseconds, not attoseconds).
* Resolve #34 by creating a "Workflow" section in the CONTRIBUTING.
* Clarify syntax for verbosity flags (see #48).
* Add badge for arXiv (see #37).
* Improved CONTRIBUTING organization.
    * Created `pull_request_template.md` as a separate file.
    * Removed templates from CONTRIBUTING - using GitHub's management of templates instead.

### CODE REVIEW

**Comment by gerudo7** [comment]
Requesting review mostly on the changes to CONTRIBUTING, since I restructured it. Are you okay with moving the code of conduct and all the templates out of the contributing file? Also, I mostly copied what you wrote on the "Workflow" section, but I did make some changes to generalize it - are those still the correct steps?

File: RELEASENOTES.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 * PR #54 Move MersenneTwister to built-in C++ function.
 	* The software now requires C++ 11 or newer.
 * PR #55 fixed sphinx documentation warnings. 
+* PR #56 Travis now checks if files are empty.
+* PR #? Improvements to documentation:
```
Comment (villaa) [inline RELEASENOTES.md:7]: Maybe reference the issue this is in reference to. I guess I forgot to. do that too in the line above. Also I think the PR is #58

File: bin/realizeCascades.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ void print_usage (FILE* stream, int exit_code)
            "  -n, --numgen        <number>       number of traces to generate \n"
            "  -o, --outfile       <filename>     name the output file \n"
            "  -s, --silent                       silent, no standard out \n"
-           "  -v, --verbose       <level>        Print verbose messages at level <level>\n"
+           "  -v, --verbose       <level>        Print verbose messages at level <level>. Currently must use `--verbose=<level>` or `-v<level>` - no spaces.\n"
```
Comment (villaa) [inline bin/realizeCascades.cpp:58]: do we need a new line or something to make this format properly? IDK I didn't check.

File: docs/source/02_Executables.rst [hunk 3]
```diff
@@ -61,7 +61,7 @@ Optional Arguments
 
 * ``-h, --help`` display the help menu. This overrides other options. Help will be displayed and program will exit. 
 * ``-s, --silent`` silent. Nothing will be printed to stdout.
-* ``-v, --verbose`` verbosity level. Default to 1 where only the random seed is printed. Max level is currently 2 where a lot of level/simulation information is printed.
+* ``-v, --verbose`` verbosity level. Default to 1 where only the random seed is printed. Max level is currently 2 where a lot of level/simulation information is printed. Currently must use `--verbose=<level>` or `-v<level>` - no spaces.
```
Comment (villaa) [inline docs/source/02_Executables.rst:64]: does this format properly?

**Review by villaa — COMMENTED** [review]
See the comments through the code--very minor. I think this is a good upgrade. The formatting of the CODE_OF_CONDUCT.md looks weird, can you look at that? It looks like it formats what should have been a list as example code??

File: bin/realizeCascades.cpp (hunk 2 above)
Comment (gerudo7) [reply bin/realizeCascades.cpp:58]: Here's the help output as it is:

```
gerudo7@Teldrassil ~/g/nrCascadeSim (bugfix/documentation-issues)> realizeCascades --help                    (root_env)
Usage:  (null) options [ inputfile(s) ]
  -d, --seed          <integer>      seed for random numbers
  -h, --help                         print usage
  -n, --numgen        <number>       number of traces to generate
  -o, --outfile       <filename>     name the output file
  -s, --silent                       silent, no standard out
  -v, --verbose       <level>        Print verbose messages at level <level>. Currently must use `--verbose=<level>` or `-v<level>` - no spaces.
  -V, --version                      print version and exit
  -l, --log           <filename>     Log additional output to the specified file. If this option is not used, no logging will occur.
Report bugs to: villaa-at-gmail-dot-com
realizeCascades (nrCascadeSim) home page: <https://github.com/villano-lab/nrCascadeSim>
General help nrCascadeSim Software: <https://nrcascadesim.readthedocs.io/en/latest/>
gerudo7@Teldrassil ~/g/nrCascadeSim (bugfix/documentation-issues)>                                           (root_env) 
```

Comment (villaa) [reply bin/realizeCascades.cpp:58]: @gerudo7 would cut off around the first sentence for "verbose" and "log" inputs; go to new line and align it with the texts starting with "Print" and "Log"

File: RELEASENOTES.md (hunk 1 above)
Comment (gerudo7) [reply RELEASENOTES.md]: Included PR # and added issue references for everything in this section.

File: bin/realizeCascades.cpp (hunk 2 above)
Comment (gerudo7) [reply bin/realizeCascades.cpp]: Done!

File: docs/source/02_Executables.rst (hunk 3 above)
Comment (gerudo7) [reply docs/source/02_Executables.rst]: Fixed.

**Comment by gerudo7** [comment]
@villaa Okay, I think I addressed everything we talked about - the only downside is that adding the sphinx requirements to the conda environment seems to have drastically increased the environment build time, depsite adding version numbers for all of the requirements. That said, it's still short enough that the tests pass.

**Comment by gerudo7** [comment]
Settling on - since most users aren't building the docs, as they're available online already, the "default" provided environment shouldn't include those dependencies at a cost to the user's time. So, I'm keeping that version of the environment file as a separate yaml (`nrCascadeSim_withDocs.yml`), and switching the main one back so that anyone who doesn't intend to build the docs can still build the environment in a reasonable amount of time.

**Pull request merged**